### PR TITLE
Fix issues parsing xml keyword in expression context

### LINKED ISSUE #29129: ParenthesisedType requires parser tests
**Description:**
ParenthesisedType tests are not available at parser. It seems some possible issues are not identified due to this. 

**Related Issues (optional):**
https://github.com/ballerina-platform/ballerina-lang/issues/26094

### LINKED ISSUE #30002: Code sample fails in the error handler
**Description:**
Following sample fails in this assert; https://github.com/ballerina-platform/ballerina-lang/blob/2ba42f45283a112cd063640c5de2fdd0e6a28d98/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java#L2033

**Sample**:
```bal
function validate(anydata value, typedesc td) {
}

public function main() {
    validate({}, xml);
    validate({}, readonly); 
}
```
```
ERROR [foo.bal:(5:18,5:18)] missing built-in type
ERROR [foo.bal:(5:21,5:22)] invalid token 'xml'
ERROR [foo.bal:(6:18,6:18)] missing identifier
ERROR [foo.bal:(6:26,6:27)] invalid token 'readonly'
```

Also re-evalutate the usage of `ERROR [foo.bal:(5:18,5:18)] missing built-in type` diagnostic.

### LINKED ISSUE #31740: Incomplete code breaks parser assumptions therefore not recovering as the optimal way
**Description:**
```bal
 function foo() {
    xml a = xml
}
```
Above test case fails at,
https://github.com/ballerina-platform/ballerina-lang/blob/8322e00a8d67114f7c0a7e28c49f020d4cf15123/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java#L2357

Therefore recovering as,
```bal
 function foo() {
    xml a =  MISSING[any] MISSING[is]xml 
 MISSING[;]}
 ```
We could rather recover this as an `xml-template-expr`.

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose
$subject.

Fixes #30002 
Fixes #31740
Fixes #29129

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -10439,8 +10447,21 @@ private STNode parseStringKeyword() {
     private STNode parseXMLTemplateExpression() {
         STNode xmlKeyword = parseXMLKeyword();
         STNode startingBackTick = parseBacktickToken(ParserRuleContext.TEMPLATE_START);
-        STNode content = parseTemplateContentAsXML();
-        STNode endingBackTick = parseBacktickToken(ParserRuleContext.TEMPLATE_END);
+
+        STNode content;
+        STNode endingBackTick;
+        if (startingBackTick.isMissing()) {
+            startingBackTick = SyntaxErrors.createMissingToken(SyntaxKind.BACKTICK_TOKEN);
```
Comment (rdulmina) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:10454]: Is this to clear the diagnostic for missing `backtick token`?

File: compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/types/ParenthesisedTypeTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package io.ballerinalang.compiler.parser.test.syntax.types;
```
Comment (rdulmina) [inline compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/types/ParenthesisedTypeTest.java:1]: missing license header

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java (hunk 1 above)
Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:10454]: Yes. that's the reason. We have done the same in `parseByteArrayLiteral()` as well.

File: compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/types/ParenthesisedTypeTest.java (hunk 2 above)
Comment (lochana-chathura) [reply compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/types/ParenthesisedTypeTest.java:18]: thanks!. updated.

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java (hunk 1 above)
Comment (rdulmina) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:10458]: Let's add a comment

Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:10459]: Added with 20e49c0

**Review by rdulmina — APPROVED** [review]

**Pull request merged**